proxy/src/main/java/com/wavefront/agent/listeners/DataDogPortUnificationHandler.java

```diff
@@ -126,6 +134,13 @@ public Long value() {
         return tagsCache.estimatedSize();
       }
     });
+    Metrics.newGauge(new TaggedMetricName("listeners", "http-relay.threadpool.queue-size",
+        "port", handle), new Gauge<Integer>() {
+      @Override
+      public Integer value() {
+        return threadpool.getQueue().size();
```

Comment:
can you check whether this is O(N)?

Reply:
that's a DelayedWorkQueue which is constant O(1) time for obtaining the size, as it is tracked internally as elements are sifted up and down internally in the BlockingQueue. Everything else about it <i>is</i> O(N) though.